developer-experience/readme.md

```diff
@@ -0,0 +1,130 @@
+# Developer Experience
+
+Developer experience refers to the how easy or difficult it is for a developer to perform essential tasks needed to implement a change. A positive developer experience would mean these tasks are relatively easy for the team (see measures below).
+
+The essential tasks are identified below.
+
+Build - Verify that changes are free of syntax error and compile.
+Test - Verify that all automated tests pass.
+Start - Launch end-to-end to simulate execution in a deployed environment.
+Debug - Attach debugger to started solution, set breakpoints, step through code, and inspect variables.
+
+If effort is invested to make these activities as easy as possible, the returns on that effort will be multiply the longer the project runs and the larger the team.
+
+## Goals
+
+- Maximize the amount of time engineers spend on writing code that fulfills story acceptance and done-done criteria.
+- Minimize the amount of time spent manual setup and configuration of tooling
+- Minimize regressions and new defects by making end-to-end testing easy
+
+## Impact
+
+Developer experience can have a significant impact on the efficiency of the day to day execution of the team. A positive experience can pay dividends throughout the lifetime of the project; especially as new developers join the team.
+
+- Increased Velocity - Team spends less time on non-value-add activities such as dev/local environment setup, waiting on remote environments to test, and rework (fixing defects).
+- Improved Quality - When its easy to debug and test, developers will do more of it. This will translate to fewer defects being introduced.
+- Easier Onboarding & Adoption - When dev essential tasks are automated, there is less documentation to write and, subsequently, less to read to get started!
+
+## Measures
+
+### Time to First Debug (aka F5 Contract)
+
+Assuming a laptop/pc that has never run the solution, how long does it take to start debugging the solution end-to-end.
+
+### Time To First Commit
+
+How long does it take to make a change that can be verified/tested locally. A locally verified/tested change is one that passes test cases without introducing regression or breaking changes.
+
+## Participation
+
+The entire development team should take responsibility in ensuring positive
+
+## Facilitation Guidance

```

Comment:
Specific examples here are fine... but let's leave room (and a clear mandate/expectation) for each team/project to define their own devex and (importantly) refine it over time via standard feedback mechanisms (retros etc.)

Reply:
Rewording the top description to this... @nmiodice i believe had similar feedback

> The following outline examples of several strategies that can be adopted to promote a positive developer experience. It is expected that each team should define what a positive dev experience means within the context of their project. Additionally, refine that over time via feedback mechanisms such as sprint and project retrospectives.